libbpf-tools/gethostlatency.bpf.c

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2021 Hengqi Chen
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include "gethostlatency.h"
+
+#define MAX_ENTRIES 10240
+
+const volatile pid_t targ_tgid = 0;
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, u32);
+	__type(value, struct val_t);
+} start SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(u32));
+	__uint(value_size, sizeof(u32));
+} events SEC(".maps");
+
+static __always_inline
+int probe_entry(struct pt_regs *ctx) {
+	if (!PT_REGS_PARM1(ctx))
+		return 0;
+
+	struct val_t val = {};
+	u32 pid = bpf_get_current_pid_tgid();
+
+	if (targ_tgid && targ_tgid != pid)
+		return 0;
+
+	if (bpf_get_current_comm(&val.comm, sizeof(val.comm)) == 0) {
+		bpf_probe_read_user(&val.host, sizeof(val.host),
```

Comment:
Check return value here too?

Reply:
Most tools in this repo do NOT check return value of `bpf_probe_read_user` and `bpf_probe_read_kernel`, how about just keep consistent with the code base ?

Reply:
Sure, I guess it doesn't hurt. An empty string doesn't hurt.